waku/waku_rln_relay/rln/wrappers.nim

```diff
@@ -161,19 +161,22 @@ proc poseidon*(data: seq[seq[byte]]): RlnRelayResult[array[32, byte]] =
   return ok(output)
 
 when defined(rln_v2):
-  func toLeaf*(rateCommitment: RateCommitment): RlnRelayResult[MerkleNode] {.inline.} =
+  proc toLeaf*(rateCommitment: RateCommitment): RlnRelayResult[seq[byte]] =
     let idCommitment = rateCommitment.idCommitment
     let userMessageLimit =  cast[array[32, byte]](rateCommitment.userMessageLimit)
-    let leafRes = poseidon(@[@idCommitment, @userMessageLimit])
-    return leafRes
-
-  func toLeaves*(rateCommitments: seq[RateCommitment]): RlnRelayResult[seq[MerkleNode]] {.inline.} =
-    var leaves = newSeq[MerkleNode](rateCommitments.len)
+    let leaf = poseidon(@[@idCommitment, @userMessageLimit]).valueOr:
+      return err("could not convert the rate commitment to a leaf")
+    var retLeaf = newSeq[byte](32)
+    for i in 0..31:
```

Comment:
perhaps something like this is safer? less magic numbers :)

```
var retLeaf = newSeq[byte](leaf.len)
for i in 0..<leaf.len:
  retLeaf[i] = leaf[i]
```

Reply:
addressed in 6f9855d5